editoast/src/views/path/projection.rs

```diff
@@ -407,111 +407,44 @@ mod tests {
-    }
-
-    #[test]
-    fn get_boundaries_case_2() {
-        let path = vec![
-            TrackRange::new("A", 50, 100, Direction::StartToStop),
-            TrackRange::new("B", 0, 200, Direction::StopToStart),
-            TrackRange::new("C", 0, 300, Direction::StartToStop),
-            TrackRange::new("D", 0, 250, Direction::StopToStart),
-            TrackRange::new("E", 25, 100, Direction::StopToStart),
-        ];
-        let projection = PathProjection::new(&path);
-
-        let track_ranges = vec![
-            TrackRange::new("X", 0, 100, Direction::StartToStop),
-            TrackRange::new("B", 0, 200, Direction::StartToStop),
-            TrackRange::new("C", 150, 200, Direction::StopToStart),
-            TrackRange::new("E", 30, 100, Direction::StartToStop),
-            TrackRange::new("Z", 0, 100, Direction::StartToStop),
-        ];
-
-        let boundaries = projection.get_intersections(&track_ranges);
-        let expected: Vec<Intersection> = vec![
-            Intersection::from((100, 350)),
-            Intersection::from((350, 420)),
-        ];
-
-        assert_eq!(boundaries, expected);
-    }
-
-    #[test]
-    fn get_boundaries_case_3() {
-        let path = vec![
-            TrackRange::new("A", 50, 100, Direction::StartToStop),
-            TrackRange::new("B", 0, 200, Direction::StopToStart),
-            TrackRange::new("C", 0, 300, Direction::StartToStop),
-            TrackRange::new("D", 0, 250, Direction::StopToStart),
-            TrackRange::new("E", 25, 100, Direction::StopToStart),
-        ];
-        let projection = PathProjection::new(&path);
-
-        let track_ranges = vec![
-            TrackRange::new("A", 0, 100, Direction::StartToStop),
-            TrackRange::new("B", 0, 200, Direction::StartToStop),
-            TrackRange::new("X", 0, 100, Direction::StartToStop),
-            TrackRange::new("C", 150, 200, Direction::StopToStart),
-            TrackRange::new("Z", 0, 100, Direction::StartToStop),
-            TrackRange::new("E", 30, 100, Direction::StartToStop),
-        ];
-
-        let boundaries = projection.get_intersections(&track_ranges);
-        let expected: Vec<Intersection> = vec![
-            Intersection::from((50, 300)),
-            Intersection::from((400, 450)),
-            Intersection::from((550, 620)),
-        ];
-
-        assert_eq!(boundaries, expected);
-    }
-
-    #[test]
-    fn get_boundaries_case_4() {
-        let path = vec![TrackRange::new("A", 0, 100, Direction::StartToStop)];
-        let projection = PathProjection::new(&path);
-
-        let track_ranges = vec![TrackRange::new("B", 0, 100, Direction::StartToStop)];
-
-        let boundaries = projection.get_intersections(&track_ranges);
-
-        let expected: Vec<Intersection> = vec![];
-        assert_eq!(boundaries, expected);
-    }
-
-    #[test]
-    fn get_boundaries_case_5() {
-        let path = vec![TrackRange::new("A", 0, 100, Direction::StartToStop)];
+    #[rstest]
```

Comment:
Could you please add a comment in the test code to explain how it works, especially the business logic part? It would help a lot in understanding it. Thanks!